server/index.js

```diff
@@ -24,6 +25,33 @@ const db = new pg.Pool({
   }
 });
 
+app.post('/api/egg', uploadsMiddleware, (req, res, next) => {
+  const message = 'message';
+  if (message in req.body) {
+    const { message, latitude, longitude, user } = req.body;
+    if (!message) throw new ClientError(400, 'message is a required field');
+    const filePath = '/images/' + req.file.filename;
+    const sql = `insert into "egg" ("message", "photoUrl", "longitude", "latitude", "creator")
+                 values ($1, $2, $3, $4, $5)
+                 returning *
+                 `;
+    const params = [message, filePath, longitude, latitude, user];
+    return db.query(sql, params)
+      .then(result => {
+        const [image] = result.rows;
+        res.json(image);
+      }).catch(err => next(err));
+  } else {
+    const sql = `
+              select *
+              from "egg"
+              `;
+    db.query(sql)
+      .then(result => res.status(200).json(result.rows))
+      .catch(err => next(err));
```

Comment:
I don't understand why this is here. You should have a separate GET route for getting the eggs (in a different feature).